table/block_based/block.cc

```diff
@@ -532,13 +532,15 @@ bool DataBlockIter::ParseNextDataKey(const char* limit) {
       // If we are reading a file with a global sequence number we should
       // expect that all encoded sequence numbers are zeros and any value
       // type is kTypeValue, kTypeMerge, kTypeDeletion, or kTypeRangeDeletion.
-      assert(GetInternalKeySeqno(key_.GetInternalKey()) == 0);
+      // Notice: Supported ingest SST files generated by live DB, so sequence
+      // and value type asserts were commented out.
+      // assert(GetInternalKeySeqno(key_.GetInternalKey()) == 0);
 
       ValueType value_type = ExtractValueType(key_.GetKey());
-      assert(value_type == ValueType::kTypeValue ||
-             value_type == ValueType::kTypeMerge ||
-             value_type == ValueType::kTypeDeletion ||
-             value_type == ValueType::kTypeRangeDeletion);
+      // assert(value_type == ValueType::kTypeValue ||
+      //        value_type == ValueType::kTypeMerge ||
+      //        value_type == ValueType::kTypeDeletion ||
+      //        value_type == ValueType::kTypeRangeDeletion);
```

Comment:
I didn't see where multiple versions of the same user key is handled. Right now I assume we generate the same key, same seqno multiple times? It might not be a good idea. What we can do here is to compare key with the previous key. If it is the same, ignore the next key.

Reply:
I think db_iter & compaction_iter can handle this.

Reply:
I am not sure that we handle same key same sequence number correctly. Even if we do today (I believe we don't), it's not an assumption we want to make in the future.